drivers/volume/pso/pso.go

```diff
@@ -0,0 +1,102 @@
+package pso
+
+import (
+	"fmt"
+	"github.com/portworx/sched-ops/k8s/core"
+	torpedovolume "github.com/portworx/torpedo/drivers/volume"
+	"github.com/portworx/torpedo/drivers/volume/portworx"
+	"github.com/portworx/torpedo/drivers/volume/portworx/schedops"
+	"github.com/portworx/torpedo/pkg/log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+const (
+	// PureDriverName is the name of the portworx-pure driver implementation
+	PureDriverName = "pso"
+	PsoServiceName = "pso-csi-controller"
+)
+
+// Provisioners types of supported provisioners
+var provisionersForPure = map[torpedovolume.StorageProvisionerType]torpedovolume.StorageProvisionerType{
+	PureDriverName: "pure-csi",
+}
+
+// pure is essentially the same as the portworx volume driver, just different in name. This way,
+// we can have separate specs for pure volumes vs. normal portworx ones
+type pso struct {
+	schedOps schedops.Driver
+	torpedovolume.DefaultDriver
+}
+
+func (d *pso) Init(sched, nodeDriver, token, storageProvisioner, csiGenericDriverConfigMap string) error {
+	log.Infof("Using the Pure volume driver with provisioner %s under scheduler: %v", storageProvisioner, sched)
+	torpedovolume.StorageDriver = PureDriverName
+	// Set provisioner for torpedo
+	if storageProvisioner != "" {
+		if p, ok := provisionersForPure[torpedovolume.StorageProvisionerType(storageProvisioner)]; ok {
+			torpedovolume.StorageProvisioner = p
+		} else {
+			return fmt.Errorf("driver %s, does not support provisioner %s", portworx.DriverName, storageProvisioner)
+		}
+	} else {
+		return fmt.Errorf("Provisioner is empty for volume driver: %s", portworx.DriverName)
+	}
+	return nil
+}
+
+func (d *pso) String() string {
+	return PureDriverName
+}
+
+func (d *pso) ValidateCreateVolume(name string, params map[string]string) error {
+	log.Warnf("ValidateCreateVolume function has not been implemented for volume driver - %s", PureDriverName)
```

Comment:
How about?

`log.Warnf("ValidateVolumeSetup function has not been implemented for volume driver - %s", d.String())`

Reply:
Add a todo like comment here informing it will be implemented in coming PRs

Reply:
Done